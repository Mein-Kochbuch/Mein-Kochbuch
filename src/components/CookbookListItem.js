import React from "react";
import styled from "styled-components/native";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function CookbookListItem({cookbook}) {

    const isDarkMode = useColorScheme() === 'dark';
    const imageSource = cookbook.thumbnail ? {uri: cookbook.thumbnail} : require("../../resources/platzhalter.png")

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <StyledImage source={imageSource}/>
            <StyledText numberOfLines={3} ellipsizeMode={"clip"}>
                {cookbook.name}
            </StyledText>
        </StyledTouchableOpacity>
    )
}

const StyledImage = styled.Image`
  height: 100px;
  width: 100px;
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`

const StyledText = styled.Text`
  max-width: 70%;
  margin-left: 12px;
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 10px;
  margin: 12px;
  display: flex;
  align-items: center;
  flex-direction: row;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
