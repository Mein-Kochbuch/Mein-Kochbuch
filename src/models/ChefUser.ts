export type ChefUser = {
  id: string;
  name: string;
};
