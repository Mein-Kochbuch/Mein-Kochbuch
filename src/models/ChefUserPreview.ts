export type ChefUserPreview = {
  id: string;
  name: string;
};
