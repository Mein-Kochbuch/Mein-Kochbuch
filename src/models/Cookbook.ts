export interface Cookbook {
  pk: number;
  name: string;
  thumbnail?: string;
}
