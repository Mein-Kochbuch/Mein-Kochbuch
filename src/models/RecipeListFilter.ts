export interface RecipeListFilter {
  sammlungsID?: number;
  count?: number;
  owner?: string;
  sortBy?: string;
  favorite?: boolean;
  maxDauer?: number;
  difficulty?: number;
  zutaten?: string[];
  searchWords?: string[];
}
