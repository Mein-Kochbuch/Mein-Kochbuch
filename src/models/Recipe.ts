import {Image} from './Image';
import {ChefUser} from './ChefUser';

export interface Recipe {
  id: string;
  title: string;
  owner: ChefUser;
  privacy: boolean;
  instruction: string;
  duration: number;
  difficulty: Difficulty;
  portions: number;
  thumbnail: Image;
  images: Image[];
  ingredients: Ingredient[];
  averageRating: number;
  ratingCount: number;
}

export type Ingredient = {
  id: string;
  amount: number;
  text: string;
};

export enum Difficulty {
  VERY_EASY = 0,
  EASY = 1,
  MEDIUM = 2,
  HARD = 3,
  EXPERT = 4,
}
