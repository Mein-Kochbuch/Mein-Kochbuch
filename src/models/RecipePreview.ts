import {Image} from './Image';
import {ChefUserPreview} from './ChefUserPreview';

export interface RecipePreview {
  id: string;
  title: string;
  averageRating: number;
  thumbnail?: Image;
  ratingCount: number;
  owner: ChefUserPreview;
}
