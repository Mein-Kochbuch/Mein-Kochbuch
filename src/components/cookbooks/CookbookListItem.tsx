import React from "react";
import styled from "styled-components/native";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";
import {Cookbook} from "../../models/Cookbook";

interface CookbookListItemProps {
    cookbook: Cookbook
}

export default function CookbookListItem({cookbook}: CookbookListItemProps) {

    const isDarkMode = useColorScheme() === 'dark';
    const imageSource = cookbook.thumbnail ? {uri: cookbook.thumbnail} : require("../../../resources/platzhalter.png")

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <StyledImage source={imageSource}/>
            <StyledText numberOfLines={2} ellipsizeMode={"middle"}>
                {cookbook.name}
            </StyledText>
        </StyledTouchableOpacity>
    )
}

const StyledImage = styled.Image`
  height: 100px;
  width: 100px;
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`

const StyledText = styled.Text`
  max-width: 70%;
  margin-left: 12px;
  text-align: center;
  font-size: 30px;
`

interface StyledTouchableOpacityProps {
    isDarkMode: boolean
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableOpacityProps>`
  padding: 12px;
  margin: 12px;
  height: 120px;
  display: flex;
  align-items: center;
  flex-direction: row;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
