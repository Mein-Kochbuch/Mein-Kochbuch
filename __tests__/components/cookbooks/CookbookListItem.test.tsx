import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import CookbookListItem from '../../../src/components/cookbooks/CookbookListItem';
import {Cookbook} from '../../../src/models/Cookbook';
import {Difficulty} from '../../../src/models/Recipe';

describe('CookbookListItem', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Snapshot Test', () => {
    const cookbook: Cookbook = {
      id: '1',
      owner: {
        id: '1',
        name: 'me',
      },
      privacy: false,
      recipes: [],
      name: 'test-title',
    };

    const component = renderer.create(<CookbookListItem cookbook={cookbook} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Snapshot with Thumbnail+Recipe', () => {
    const cookbook: Cookbook = {
      id: '1',
      owner: {
        id: '1',
        name: 'me',
      },
      privacy: false,
      recipes: [
        {
          title: 'recipe-1',
          id: '1',
          averageRating: 3.5,
          images: [],
          instruction: 'instructions-1',
          difficulty: Difficulty.EXPERT,
          portions: 4,
          privacy: false,
          duration: 15,
          ratingCount: 13,
          thumbnail: {
            id: '1',
            url: 'thumnail-url-1',
          },
          ingredients: [
            {
              id: '1',
              amount: 500,
              text: 'ingredient-1',
            },
          ],
          owner: {name: 'me', id: '1'},
        },
      ],
      name: 'test-title',
      thumbnail: {
        id: '1',
        url: 'test-uri',
      },
    };

    const component = renderer.create(<CookbookListItem cookbook={cookbook} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
