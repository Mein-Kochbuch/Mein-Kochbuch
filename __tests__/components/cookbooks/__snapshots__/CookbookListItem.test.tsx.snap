// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookbookListItem Snapshot Test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#222",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "height": 120,
      "marginBottom": 12,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 12,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../resources/platzhalter.png",
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "#68a0cf",
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 100,
          "paddingBottom": 20,
          "paddingTop": 20,
          "width": 100,
        },
      ]
    }
  />
  <Text
    ellipsizeMode="middle"
    numberOfLines={2}
    style={
      Array [
        Object {
          "fontSize": 30,
          "marginLeft": 12,
          "maxWidth": "70%",
          "textAlign": "center",
        },
      ]
    }
  >
    test-title
  </Text>
</View>
`;

exports[`CookbookListItem Snapshot with Thumbnail+Recipe 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#222",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "height": 120,
      "marginBottom": 12,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 12,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "test-uri",
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "#68a0cf",
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 100,
          "paddingBottom": 20,
          "paddingTop": 20,
          "width": 100,
        },
      ]
    }
  />
  <Text
    ellipsizeMode="middle"
    numberOfLines={2}
    style={
      Array [
        Object {
          "fontSize": 30,
          "marginLeft": 12,
          "maxWidth": "70%",
          "textAlign": "center",
        },
      ]
    }
  >
    test-title
  </Text>
</View>
`;
